packages/smooth_app/lib/pages/scan/continuous_scan_page.dart

```diff
@@ -1,114 +1,140 @@
 import 'package:flutter/material.dart';
-import 'package:flutter_gen/gen_l10n/app_localizations.dart';
 import 'package:flutter_svg/flutter_svg.dart';
 import 'package:provider/provider.dart';
 import 'package:qr_code_scanner/qr_code_scanner.dart';
 import 'package:smooth_app/data_models/continuous_scan_model.dart';
 import 'package:smooth_app/pages/personalized_ranking_page.dart';
 import 'package:smooth_app/widgets/smooth_product_carousel.dart';
 import 'package:smooth_ui_library/smooth_ui_library.dart';
+import 'package:smooth_ui_library/util/ui_helpers.dart';
 
 class ContinuousScanPage extends StatelessWidget {
   final GlobalKey _scannerViewKey = GlobalKey(debugLabel: 'Barcode Scanner');
 
   @override
   Widget build(BuildContext context) {
-    final ContinuousScanModel model = context.watch<ContinuousScanModel>();
-    final AppLocalizations localizations = AppLocalizations.of(context)!;
-    final Size screenSize = MediaQuery.of(context).size;
-    final double carouselHeight = screenSize.height / 2;
-    final double viewFinderBottomOffset = carouselHeight / 2.0;
-    return Scaffold(
-      appBar: AppBar(toolbarHeight: 0.0),
-      body: Stack(
-        children: <Widget>[
-          Container(
-            alignment: Alignment.center,
-            color: Colors.black,
-            child: Padding(
-              // Double the offset to account for the vertical centering.
-              padding: EdgeInsets.only(bottom: 2 * viewFinderBottomOffset),
-              child: SvgPicture.asset(
-                'assets/actions/scanner_alt_2.svg',
-                width: 60.0,
-                height: 60.0,
-                color: Colors.white,
+    return LayoutBuilder(
+        builder: (BuildContext context, BoxConstraints constraints) {
+      final Size screenSize = MediaQuery.of(context).size;
+      final Size scannerSize = Size(
+        screenSize.width * 0.6,
+        screenSize.width * 0.33,
+      );
+      final ContinuousScanModel model = context.watch<ContinuousScanModel>();
+      final double carouselHeight =
+          constraints.maxHeight / 1.81; // roughly 55% of the available height
+      final double buttonRowHeight = areButtonsRendered(model) ? 48 : 0;
+      final double availableScanHeight =
+          constraints.maxHeight - carouselHeight - buttonRowHeight;
+      // Padding for the qr code scanner. This ensures the scanner has equal spacing between buttons and carousel.
+      final EdgeInsets qrScannerPadding = EdgeInsets.only(
+          top:
+              (availableScanHeight - scannerSize.height) / 2 + buttonRowHeight);
+      final double viewFinderBottomOffset = carouselHeight / 2.0;
```

Comment:
not entirely sure if this is needed.... (?)

Reply:
It was added by @slava-sh a while ago, but I don't know what it's exact purpose is

Reply:
I will try and test on an iOS device, (device_preview renders fine without this offset but maybe there's some issue on iOS)